### DISCUSSION TITLE
Writing Login info to keychain failed: StartServiceByName for org.freedesktop.secrets timeout reached in VSCode

### DISCUSSION THREAD

**Discussion by jarmst95** [community/community#7115] [Copilot Conversations]
After authorizing copilot in VS Code I get this error
```
Writing login information to the keychain failed with error 'Error calling StartServiceByName for org.freedesktop.secrets: Timeout was reached'.
```
Previously I was getting an error to which I found a solution here https://github.com/microsoft/vscode/issues/120392#issuecomment-814210643

I'm using the same `.xinitrc` as given in that solution (except I launch xmonad instead of dwm and I tried with and without the gpg component since the man page for `gnome-keyring-daemon` on my system doesn't list it as a component.)

I  have gnome-keyring and seahorse installed.

```
$ cat /etc/os-release
NAME="Ubuntu"
VERSION="20.04.3 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.3 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
```

```
$ code --version
1.61.2
6cba118ac49a1b88332f312a8f67186f7f3c1643
x64  
```
copilot version 1.6.3571

**Comment by alanibrus** [comment]
I do not have the same setup or exactly the same issue, but I was struggling with same type of error for nextcloud after latest system upgrade. But after hours of trying to get a fix and also going through all sorts of different .xinitrc's, eventually all I had to do was to change:

`dbus-launch startxfce4` to `exec startxfce4` in .xinitrc.

Maybe it helps in your case, maybe not.

**Reply by vinilshah1** [reply]
Thank you so much, this worked!

**Comment by jarmst95** [comment]
@alanibrus  When change back from `dbus-launch` to just `exec` I get this error `Writing login information to the keychain failed with error 'Unknown or unsupported transport “disabled” for address “disabled:”'.` per this issue: https://github.com/microsoft/vscode/issues/120392

**Reply by LiteApplication** [reply]
What about `exec dbus-launch startxfce4` ?

**Comment by norandom** [comment]
I have the same issue on Ubuntu

**Reply by shadycuz** [reply]
same =/

**Comment by frohro** [comment]
Same here on Ubuntu 22.10.  =/

**Comment by holms** [comment]
No matter what I've tried I'm getting same errors either ```Writing login information to the keychain failed with error 'Error calling StartServiceByName for org.freedesktop.secrets: Timeout was reached'.```

or 

```Writing login information to the keychain failed with error 'Unknown or unsupported transport “disabled” for address “disabled:”'.```

Why can't we use just simple config files? I have DWM in here I don't need gnome at all.

**Comment by darkoverlordofdata** [comment]
I'm getting this error on FreeBSD:
```
Writing login information to the keychain failed with error 'Error calling StartServiceByName for org.freedesktop.secrets: Process org.freedesktop.secrets exited with status 1'.
```

The thing is, it actually works in that my changes are posted to GitHub., so I wonder, what is the actual error? Is it safe to use anyway? What exactly is it trying to do? It doesn't seem to have any effect by failing.

**Comment by GrabbenD** [comment]
Have anyone found the answer to this?

Fixed it: https://gist.github.com/GrabbenD/6658c36a1c7fc7ee30ee2498647ca4c6
